src/htsp_server.c

```diff
@@ -1423,14 +1441,20 @@ htsp_method_async(htsp_connection_t *htsp, htsmsg_t *in)
 
   /* Send EPG updates */
   if (epg) {
```

Comment:
This code block appears almost same as htsp_epg_window_cb(). I would create a fcn with the shared code called from here and from htsp_epg_window_cb().